babylon-to-espree/toAST.js

```diff
@@ -94,7 +94,11 @@ var astTransformVisitor = {
     if (path.isRegExpLiteral()) {
       node.type = "Literal";
       node.raw = node.extra.raw;
-      node.value = {};
+      try {
+        node.value = new RegExp(node.pattern, node.flags);
+      } catch (err) {
+        node.value = null;
```

Comment:
A scenario where is is null is when the flags `u` and `y` are not supported. I have tests cases for these and it will produce null when run in node v4 and v5. 

I'm not sure this is a breaking change since the output is was producing is a bug and did not match what espree, or esprima produce. The reason the test pass is it only compared the `typeof`, which were both objects, but not from the same class. We could set the default to `{}` but that would not match the ESTree spec for RegExp Literal nodes.
https://github.com/estree/estree/blob/master/es5.md#regexpliteral

This does fix an issue with eslint https://github.com/eslint/eslint/issues/8655 and setting the default to `{}` might cause other rule failures.

The estree plugin for babylon does produce the correct output so if this were to be considered a breaking change we could wait till that is implemented in this parser.